### PR TITLE
Cancel request by timeout

### PR DESCRIPTION
**Pull request by g4s8**
#409 - wrapped `Pie` slices with `TimeoutSlice`

### CODE REVIEW

File: src/main/java/com/artipie/http/TimeoutSlice.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            this.exec.schedule(
+                () -> {
+                    if (this.completed.compareAndSet(false, true)) {
+                        this.origin.accept(
+                            RsStatus.UNAVAILABLE, Headers.EMPTY,
+                            new Content.From(
+                                "Request cancelled by timeout".getBytes(StandardCharsets.UTF_8)
+                            )
+                        );
+                        future.complete(null);
```
Comment (olegmoz) [inline src/main/java/com/artipie/http/TimeoutSlice.java:205]: @g4s8 `future` should complete only after `this.origin.accept` has been completed, please move `future.complete` invocation to `.thenRun()`

File: src/main/java/com/artipie/http/TimeoutSlice.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+                    }
+                },
+                duration.getSeconds(), TimeUnit.SECONDS
```
Comment (olegmoz) [inline src/main/java/com/artipie/http/TimeoutSlice.java:208]: @g4s8 I'd suggest to use milliseconds here as it is default precision in Java libraries

**Comment by g4s8** [comment]
@olegmoz thanks, should be fixed now: https://github.com/artipie/artipie/pull/412/commits/79a58b6ead915bbbdee277a7bb757a0d94a8303f

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#412` is not assigned, can't get performer

**Pull request merged**